### ISSUE TITLE
 About custome dataset

### ISSUE THREAD

**Issue by cyy21** [PeizeSun/SparseR-CNN#9]
do you have any suggestions on how to train on a custom dataset? Is sparse r-cnn need at least ~15k images to converge like detr, will sparse r-cnn work on a small dataset?

**Comment by PeizeSun** [comment]
Hi~
I am pretty sure that sparse r-cnn works on a small dataset. I've used sparse r-cnn on a small dataset, where the training set only contains ~2k images, and it works well.
You can registry your dataset in detectron2, change config file of sparse r-cnn and train your model.

**Comment by cyy21** [comment]
@PeizeSun Thanks for reply, and dou you think change the bbox to polygon will work, the label of my dataset is not xywh

**Comment by PeizeSun** [comment]
I guess it will work. You can initialize the proposal box as proposal polygon, use the circumscribed rectangle of the polygon to extract RoI feature,  use regression branch to predict refined polygon.

**Comment by cyy21** [comment]
@PeizeSun hi, i just test with 1 img to see if model can overfit it, and fails. Is it normal,should i train with the full dataset?

**Comment by cyy21** [comment]
the output is all same, i met the same problem in detr, is it the problem caused by bipartite graph matching？

**Comment by PeizeSun** [comment]
If you train with only 1  image, I suggest you choose a very small learning rate to see wether it works.
